### ISSUE TITLE
[bug] Box upgrade in 3.1.0 causes DynaBox signature change and  broke existing code

### ISSUE THREAD

**Issue by ap--** [rochacbruno/dynaconf#391]
Hi @rochacbruno 

**Describe the bug**

I was using the following code in an existing module, and the 3.1.0 release causes an error due to 
0e4752a979be8cd5f872a06d049886e589dc7da8 (59be115724f9d973229bc8e1b884a3ae3e4f9bd5) updating Box, which now requires the `box_settings` keyword argument.

```python
>       data = DynaBox(settings.as_dict(internal=False))
E       TypeError: __new__() missing 1 required keyword-only argument: 'box_settings'
```

**To Reproduce**

```console
pip install dynaconf==3.1.0
```

```python
# contents of error.py
from dynaconf import settings
from dynaconf.utils.boxing import DynaBox

data = DynaBox(settings.as_dict())
```

```python
# Traceback python error.py
(dynaconf) ➜  dynaconf git:(master) ✗ python error.py 
Traceback (most recent call last):
  File "error.py", line 5, in <module>
    data = DynaBox(settings.as_dict())
TypeError: __new__() missing 1 required keyword-only argument: 'box_settings'
```

```console
(dynaconf) ➜  dynaconf git:(master) ✗ pip list
Package    Version
---------- -------
dynaconf   3.1.0
pip        20.1
setuptools 46.1.3
wheel      0.34.2
```

**Expected behavior**

Should I add a default argument for `box_settings` for dynaconf versions `>=3.1.0`? Is `DynaBox` considered private? Or is this a bug and will `DynaBox` provide a default argument for `box_settings` in `Box.__new__`?

Cheers,
Andreas 😃

**Comment by rochacbruno** [comment]
@RonnyPfannschmidt is that the same thing you mentioned about?

**Comment by RonnyPfannschmidt** [comment]
No, i can link you to a back trace once I return to the computer

**Comment by JacobCallahan** [comment]
This may be related to what I just hit.
```
...
 File "/home/jake/Programming/venvbroker/lib64/python3.8/site-packages/broker/settings.py", line 3, in <module>
    from dynaconf import Dynaconf, Validator
  File "/home/jake/Programming/venvbroker/lib64/python3.8/site-packages/dynaconf/__init__.py", line 6, in <module>
    from dynaconf.base import LazySettings  # noqa
  File "/home/jake/Programming/venvbroker/lib64/python3.8/site-packages/dynaconf/base.py", line 9, in <module>
    from dynaconf import default_settings
  File "/home/jake/Programming/venvbroker/lib64/python3.8/site-packages/dynaconf/default_settings.py", line 10, in <module>
    from dynaconf.utils.parse_conf import parse_conf_data
  File "/home/jake/Programming/venvbroker/lib64/python3.8/site-packages/dynaconf/utils/parse_conf.py", line 11, in <module>
    from dynaconf.utils.boxing import DynaBox
  File "/home/jake/Programming/venvbroker/lib64/python3.8/site-packages/dynaconf/utils/boxing.py", line 6, in <module>
    from dynaconf.vendor.box import Box
  File "/home/jake/Programming/venvbroker/lib64/python3.8/site-packages/dynaconf/vendor/box/__init__.py", line 7, in <module>
    from box.box import Box
ModuleNotFoundError: No module named 'box.box'; 'box' is not a package
```

**Comment by RonnyPfannschmidt** [comment]
thats it

**Comment by JacobCallahan** [comment]
@rochacbruno interestingly upgrading python-box seems like it fixed the above issue. `python-box-5.1.0`

**Comment by RonnyPfannschmidt** [comment]
i suspect a unintended problem with the vendoring mechanism, but i dont have enough time to substantiate the hunch

**Comment by ap--** [comment]
Hi everyone,

I'm not sure if I got my main issue across correctly. I think I chose a wrong title, because I was trying to over-analyse what's happening here.

My actual issue is:
with `dynaconf==3.0.0` I could instantiate `DynaBox` without providing `box_settings` and now it's required. I was assuming that `DynaBox` is considered public and was not expecting that it would change its required call arguments or public interface with a minor version increase.

I hope I didn't cause too much confusion. Sorry for that 🙇 😊 

Cheers,
Andreas 😃
